src/com/projectkorra/projectkorra/event/SimulatedBlockPlaceEvent.java

```diff
@@ -0,0 +1,20 @@
+package com.projectkorra.projectkorra.event;
+
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.entity.Player;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+
+/**
+ * Called as a test event for checking if other plugins cancel the block placing
+ * @see {@link com.projectkorra.projectkorra.GeneralMethods#isRegionProtectedFromBuildPostCache}
+ */
+public class SimulatedBlockPlaceEvent extends BlockPlaceEvent {
+
+	public SimulatedBlockPlaceEvent(Block placedBlock, BlockState replacedBlockState, Block placedAgainst, ItemStack itemInHand, Player thePlayer, boolean canBuild, EquipmentSlot hand) {
+		super(placedBlock, replacedBlockState, placedAgainst, itemInHand, thePlayer, canBuild, hand);

```

Comment:
Since this is our own event we probably want to consider changing the parameters to fit our needs rather than just copying what's already there. Most of the information for our version of the event can be obtained from a Location and Player parameter, with `canBuild` being true and `hand` being EquipmentSlot.HAND. Also can consider passing in a custom item into the super call so that nothing funky happens if maybe another plugin uses the item in the player's hand and the place event.

Reply:
Made changes; I'm not sure what kind of custom item we would use though.

Reply:
Used petrified oak slab at the suggestion of @ChristopherWMM. Let me know if you object.

Reply:
Made item static